src/serialize.rs

```diff
@@ -91,7 +91,7 @@ pub fn node_to_stream(node: &Node, f: &mut dyn io::Write) -> io::Result<()> {
 /// decode the length prefix for an atom. Atoms whose value fit in 7 bits
 /// don't have a length prefix, so those should be handled specially and
 /// never passed to this function.
-fn decode_size(f: &mut dyn io::Read, initial_b: u8) -> io::Result<u64> {
+fn decode_size(f: &mut Cursor<&[u8]>, initial_b: u8) -> io::Result<u64> {
```

Comment:
What do you think of this instead:

```suggestion
fn decode_size<R: io::Read>(f: &mut R, initial_b: u8) -> io::Result<u64> {
```

Requires fewer changes and should be just as fast, since it's a template. I may want to use the `Read` trait in future.